scripts/load_meeting_prot.py

```diff
@@ -21,91 +24,90 @@ def get_ws(wsheet_title,ssheet):
-                    if row.strip() not in data[proj]['info']:
-                        data[proj]['info'] = data[proj]['info']+' - '+row
-            elif proj and row.split()[0][0]=='1':
-                if not data[proj]['flowcells'].has_key(row.strip()):
-                    data[proj]['flowcells'][row.strip()]=[]
-                proj=None
-            i=i+1
-        else:
-            i=i+1
+    dict_holder = coming_deliveries
+    proj_name = None
+    for row in content:
+        row = row.strip()
+        if row == "Ongoing":
+            dict_holder = ongoing_deliveries
+        if row:
+            if len(row) > 2 and row[1] == '.' or row[2] == '.':
+                row_list = row.split()
+                proj_name = row_list[0]
+                info = ' '.join(row_list[1:])
+                if not proj_name in dict_holder.keys():
+                    dict_holder[proj_name] = {'info' : info, 'flowcells' : {}}
+            elif proj_name and not row[0].isdigit():
+                if dict_holder[proj_name].has_key('info'):
+                    if row not in dict_holder[proj_name]['info']:
+                        info = dict_holder[proj_name]['info']
+                        more_info = ' - '.join([info, row])
+                        dict_holder[proj_name]['info'] = more_info
+            elif proj_name and row[0].isdigit():
+                if not dict_holder[proj_name]['flowcells'].has_key(row):
+                    dict_holder[proj_name]['flowcells'][row] = []
+                proj_name = None
     return coming_deliveries, ongoing_deliveries
 
 def update(sorted_names, col, info, ss_key, ws_key):
     """Uppdates the new meeting protocol with old and new info."""
-    i=2
-    for name in sorted_names:
-        name = name[0]
-        CLIENT.UpdateCell(i, col, name , ss_key, ws_key)
-        CLIENT.UpdateCell(i, col+1, info[name]['info'] , ss_key, ws_key)
-        fcs = sorted(info[name]['flowcells'].keys())
-        for fc in fcs:
-            i=i+1
-            j=i
-            CLIENT.UpdateCell(i, col+1, fc, ss_key, ws_key)
-            for comment in info[name]['flowcells'][fc]:
-                CLIENT.UpdateCell(j, col+2, comment, ss_key, ws_key)
-                j=j+1
-            if i!=j:
-                i=j-1
-        i=i+2
+    row = 2
+    for proj_name in sorted_names:
+        proj_name = proj_name[0]
+        CLIENT.UpdateCell(row, col, proj_name , ss_key, ws_key)
+        CLIENT.UpdateCell(row, col+1, info[proj_name]['info'] , ss_key, ws_key)
+        sorted_fcs = sorted(info[proj_name]['flowcells'].keys())
+        for fc in sorted_fcs:
+            row += 1
+            comments_row = row
+            CLIENT.UpdateCell(row, col+1, fc, ss_key, ws_key)
+            for comment in info[proj_name]['flowcells'][fc]:
+                CLIENT.UpdateCell(comments_row, col+2, comment, ss_key, ws_key)
+                comments_row += 1
+            if row != comments_row:
+                row = comments_row - 1
```

Comment:
Yes! Lest do it next time we meet! 

/M
  
On Jan 28, 2015, at 9:17 AM, Francesco <notifications@github.com> wrote:

> In scripts/load_meeting_prot.py:
> 
> > -        i=i+2
> > +    row = 2
> > +    for proj_name in sorted_names:
> > +        proj_name = proj_name[0]
> > +        CLIENT.UpdateCell(row, col, proj_name , ss_key, ws_key)
> > +        CLIENT.UpdateCell(row, col+1, info[proj_name]['info'] , ss_key, ws_key)
> > +        sorted_fcs = sorted(info[proj_name]['flowcells'].keys())
> > +        for fc in sorted_fcs:
> > +            row += 1
> > +            comments_row = row
> > +            CLIENT.UpdateCell(row, col+1, fc, ss_key, ws_key)
> > +            for comment in info[proj_name]['flowcells'][fc]:
> > +                CLIENT.UpdateCell(comments_row, col+2, comment, ss_key, ws_key)
> > +                comments_row += 1
> > +            if row != comments_row:
> > +                row = comments_row - 1
> @mayabrandi I really want to sit with you and discuss this for loop
> 
> —
> Reply to this email directly or view it on GitHub.
>